unit_tests/environment/test_battle.py

```diff
@@ -405,6 +405,56 @@ def test_battle_request_and_interactions(example_request):
     assert not battle.maybe_trapped
     assert battle.opponent_can_dynamax
 
+    # Items
+    battle._parse_message(
+        ['', '-damage', 'p2a: Necrozma', '167/319', '[from] item: Rocky Helmet', '[of] p1a: Groudon']
+    )
+    assert battle.opponent_active_pokemon.item == "rockyhelmet"
+    battle.opponent_active_pokemon._item = None
```

Comment:
This is one long test, so I reset the item/ability changes like this to make sure a later test doesn't assert something that happened in a previous test. Need to be careful when asserting multiple things in one test :slightly_smiling_face: